File: app/src/main/java/com/ambinusian/adab/MainActivity/CourseRecyclerView/CourseAdapter.java
package com.ambinusian.adab.MainActivity.CourseRecyclerView;

import android.content.Context;
import android.provider.ContactsContract;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.ambinusian.adab.R;

import java.util.ArrayList;

public class CourseAdapter extends RecyclerView.Adapter<CourseHolder> {

    private Context context;
    private ArrayList<CourseModel> list;

    public CourseAdapter(Context context, ArrayList<CourseModel> list){
        this.context = context;
        this.list = list;
    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    @NonNull
    @Override
    public CourseHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.item_layout_courses,parent,false);
        return new CourseHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull CourseHolder holder, int position) {
        CourseModel item = list.get(position);
        holder.classTime.setText(item.getClassTime());
        holder.classTitle.setText(item.getClassTitle());
        holder.courses.setText(item.getCourse());
        holder.classMeeting.setText(item.getClassMeeting());
        holder.courseCode.setText(item.getCourseCode());
        holder.classCode.setText(item.getClassCode());
        holder.classType.setText(item.getClassType());
    }
}

File: app/src/main/java/com/ambinusian/adab/MainActivity/CourseRecyclerView/CourseAdapter.kt
package com.ambinusian.adab.MainActivity.CourseRecyclerView

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.ambinusian.adab.R

class CourseAdapter(private val courseList : ArrayList<CourseModel>) : RecyclerView.Adapter<CourseHolder>(){
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CourseHolder {
        val layout = LayoutInflater.from(parent.context).inflate(R.layout.item_layout_courses, parent, false)
        return CourseHolder(layout)
    }

    override fun getItemCount(): Int {
        return courseList.size
    }

    override fun onBindViewHolder(holder: CourseHolder, position: Int) {
        if(position < courseList.size){
            val list = courseList.get(position)
            holder.courseTime.text = list.courseTime
            holder.courseTitle.text = list.courseTitle
            holder.course.text = list.course
            holder.courseCode.text = list.courseCode
            holder.courseMeeting.text = list.courseMeeting
            holder.classCode.text = list.classCode
            holder.classType.text = list.classType
        }
    }
}